hydradx/notebooks/Omnipool/TestPriceMoveWithdraw.ipynb

```diff
@@ -0,0 +1,585 @@
+    "        assets['R1']['weight'] = weight\n",
+    "        assets['R2']['weight'] = 0.75 - (weight * 2)\n",
+    "        # assert round(sum([assets[tkn]['weight'] for tkn in assets]), 10) == 1.0\n",
+    "        \n",
+    "        for tkn, info in assets.items():\n",
+    "            liquidity[tkn] = initial_omnipool_tvl * info['weight'] / info['usd price']\n",
+    "            lrna[tkn] = initial_omnipool_tvl * info['weight'] / lrna_price_usd\n",
+    "\n",
+    "        initial_state = GlobalState(\n",
+    "            pools={\n",
+    "                'omnipool': OmnipoolState(\n",
+    "                    tokens={\n",
+    "                        tkn: {'liquidity': liquidity[tkn], 'LRNA': lrna[tkn]} for tkn in assets\n",
+    "                    },\n",
+    "                    lrna_fee=0.0005,\n",
+    "                    asset_fee=0.0025\n",
+    "                )\n",
+    "            },\n",
+    "            agents = {\n",
+    "                'Arbitrageur': Agent(\n",
+    "                    holdings={tkn: 10000000000 for tkn in list(assets.keys()) + ['LRNA']},\n",
+    "                    trade_strategy=omnipool_arbitrage(\n",
+    "                        pool_id='omnipool',\n",
+    "                        arb_precision=1\n",
+    "                    )\n",
+    "                ),\n",
+    "                'LP': Agent(\n",
+    "                    holdings={'R1': liquidity['R1']},\n",
+    "                    trade_strategy=invest_all(\n",
+    "                        pool_id='omnipool'\n",
+    "                    ) + withdraw_all(time_steps/2)\n",
+    "                )\n",
+    "            },\n",
+    "            evolve_function=historical_prices(price_list=[\n",
+    "                {\n",
+    "                    tkn: assets[tkn]['usd price'] * (pump_factor - (abs(i * 2 - (time_steps - 2)) / (time_steps - 2) * (pump_factor - 1)))\n",
+    "                    for tkn in price_move_assets\n",
+    "                } \n",
+    "                for i in range(time_steps - 1)\n",
+    "            ] + [{tkn: assets[tkn]['usd price'] for tkn in price_move_assets}] * 2),\n",
+    "            external_market={tkn: assets[tkn]['usd price'] for tkn in assets}\n",
+    "        )\n",
+    "        events.append(run.run(initial_state, time_steps=time_steps, silent=True)[-1])\n",
+    "    \n",
+    "    plt.figure(figsize=(20, 5))\n",
+    "    plt.plot(range(len(events)), [1 / x for x in pu.get_datastream(events, pool='omnipool', prop='lrna_price', key='USD')])\n",
+    "    plt.xlabel('weight of TKN in omnipool')\n",
+    "    plt.ylabel('LRNA price in USD')\n",
+    "    ticks_number = len(tkn_weights[::10])\n",
+    "    plt.xticks(\n",
+    "        [1 / (ticks_number - 1) * i * len(events) for i in range(ticks_number)],\n",
+    "        [f'{int(i * 100)}%' for i in tkn_weights[::10]]\n",
+    "    )\n",
+    "    plt.title(f\"price move: {price_moves[pump_factor]}\")\n",
```

Comment:
Why does this graph look like a sawtooth graph instead of a smooth one? Doesn't make sense to me...

Reply:
This isn't the solution that's currently checked in, is it?